modules/exploits/multi/fileformat/libreoffice_macro_exec.rb

```diff
@@ -0,0 +1,113 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::FILEFORMAT
+  include Msf::Exploit::Powershell
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'LibreOffice Macro Code Execution',
+      'Description'    => %q{
+        LibreOffice comes bundled with sample macros written in Python and
+        allows the ability to bind program events to them. A macro can be tied
+        to a program event by including the script that contains the macro and
+        the function name to be executed. Additionally, a directory traversal
+        vulnerability exists in the component that references the Python script
+        to be executed. This allows a program event to execute functions from Python
+        scripts relative to the path of the samples macros folder. The pydoc.py script
+        included with LibreOffice contains the tempfilepager function that passes
+        arguments to os.system, allowing RCE.
+
+        This module generates an ODT file with a mouse over event that
+        when triggered, will execute arbitrary code.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+      [
+        'Alex Inführ', # Vulnerability discovery and PoC
+        'Shelby Pace'  # Metasploit Module
+      ],
+      'References'     =>
+        [
+          [ 'CVE', '2018-16858' ],
+          [ 'URL', 'https://insert-script.blogspot.com/2019/02/libreoffice-cve-2018-16858-remote-code.html' ]
+        ],
+      'Platform'       => [ 'win', 'linux' ],
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
```

Comment:
While 'Arch' here seems to indicate I can select a 64-bit payload, the targets below restrict this to just 32-bit.